ChatSX/screens/ChatRoom/index.tsx

```diff
@@ -2329,18 +2479,228 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
     const isScrollAtEnd =
       contentLength - visibleLength - offset < onEndReachedThreshold;
 
-    if (isScrollAtStart && shouldLoadMoreChatStart && lastScrollOffset) {
+    if (isScrollAtStart && shouldLoadMoreChatStart && !isFound) {
       renderFooter();
       onStartReached();
-      setLastScrollOffset(false);
-      setTimeout(() => {
-        setLastScrollOffset(true);
-      }, 1000);
     }
 
     if (isScrollAtEnd && shouldLoadMoreChatEnd) {
       renderFooter();
-      onEndReached();
+      handleLoadMore();
+    }
+  };
+
+  const getIconAttachment = (val: any) => {
```

Comment:
move to this whole functions into `ChatroomUtils` and divide it into multiple smaller functions

Reply:
I have divided this `getIconAttachment

Reply:
I have divided the `getIconAttachment` into multiple methods based on the `if` conditions but not able to move it to a separate utils file as it contains tags like `<View><Text>` etc which can only be used inside a functional component otherwise it throws error